### PR TITLE
Add more unit tests for devicemanager/endpoint.

### PR DESCRIPTION
**Pull request by tizhou86**
**What this PR does / why we need it**:
Add more unit tests for devicemanager/endpoint.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
NONE

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/cm/devicemanager/endpoint_test.go [hunk 1]
```diff
@@ -169,6 +169,40 @@ func TestAllocate(t *testing.T) {
...
+	}
+	callbackCount := 0
+	callbackChan := make(chan int)
+	p, e := esetup(t, devs, socket, "mock", func(n string, a, u, r []pluginapi.Device) {
+		callbackCount++
+		callbackChan <- callbackCount
+	})
+	defer ecleanup(t, p, e)
+
+	resp := new(pluginapi.PreStartContainerResponse)
```
Comment (RenaudWasTaken) [inline pkg/kubelet/cm/devicemanager/endpoint_test.go]: &pluginapi.PreStartContainerResponse{}

File: pkg/kubelet/cm/devicemanager/endpoint_test.go [hunk 2]
```diff
@@ -169,6 +169,40 @@ func TestAllocate(t *testing.T) {
...
+	select {
+	case <-callbackChan:
+		break
+	case <-time.After(time.Second):
+		t.FailNow()
+	}
+
+	respOut, err := e.preStartContainer([]string{"PreStartDeviceId-1", "PreStartDeviceId-2"})
+	require.NoError(t, err)
+	require.Equal(t, resp, respOut)
```
Comment (RenaudWasTaken) [inline pkg/kubelet/cm/devicemanager/endpoint_test.go]: Looks like we are testing if the value is empty, I'm wondering if it wouldn't be better to have the same pattern as for the callback (i.e: make sure the function is called.

**Review by RenaudWasTaken — COMMENTED** [review]
Might be good to test the whole code path (i.e: Test from manager)

File: pkg/kubelet/cm/devicemanager/endpoint_test.go (hunk 2 above)
Comment (tizhou86) [reply pkg/kubelet/cm/devicemanager/endpoint_test.go]: Sure thing, I'll add more test scenario tomorrow.

Comment (tizhou86) [reply pkg/kubelet/cm/devicemanager/endpoint_test.go]: It seems to be currently PreStartContainerResponse's protobuf is empty, what should be the meaningful struct for this response?

**Comment by tizhou86** [comment]
Close this pr atm.

**Pull request closed without merging**